esmvalcore/_recipe.py

```diff
@@ -910,6 +943,9 @@ def _initialize_variables(self, raw_variable, raw_datasets):
             variable = deepcopy(raw_variable)
             variable.update(dataset)
             variable['recipe_dataset_index'] = index
+            # preserve the MIP from RAW in variable object
+            if 'mip' in raw_variable:
```

Comment:
I'm not too keen on this, it makes the behaviour more difficult to understand. Now it's use whatever is in variable and overwrite with dataset. This adds an exception that can be confusing and is not needed.

Reply:
moreover for CMIP6 when fx mips differ so much pretty much eveything mip related will have to stay in variable and not in datasets

Reply:
Most of the users are used to the current behaviour and changing it without obvious advantages could be dangerous at this stage.

Reply:
By far the most logical place for the mip is with the variable, because it defines the variable: i.e. project/mip/variable is how variables are defined in the CMOR tables.

>  so even if they have 99 variables with the same mip and only one with a different mip they'll have to put the mip under each of the variables

Yes, but they can use yaml anchors to avoid repeating themselves.

Reply:
ok but then variables that dont have the mip from the datasets will not work without this

Reply:
"most logical place for the mip is with the variable..." -> exactly, but if the mip is in `datasets` that'll get reset and we're back in the dungeons

Reply:
It has worked fine so far

Reply:
it doesn't, man! look at your code :grin:

Reply:
Both approaches have pros and cons, the big advantage of the current one is that everybody is used to it.

Reply:
```
datasets:
  - {dataset: bcc-csm1-1,  project: CMIP5, mip: Amon, exp: historical,  ensemble: r1i1p1,  start_year: 2000,  end_year: 2002}

diagnostics:
  example:
    description: Example diagnostic
    themes:
      - phys
    realms:
      - atmos
    variables:
      ta:
        preprocessor: preprocessor_1
        reference_dataset: ERA-Interim
      sftlf:
        mip: fx
```
will automatically assign `Amon` to `sftlf`

Reply:
all I'm saying is that if the user wants to declare the mip under their variable, automatic assignment from `datasets` should be disabled, it's a pro option or whatevs you wanna call it, it doesn't turn off the previous functionality that they're used to. It's like an override button

Reply:
This should be the last open issue on this PR.
Any further thoughts?
To stay on the safe side, I would suggest **not** to change this behaviour now.

Reply:
OK so if yous are so much against this then we'll have to make sure we include a very **clear** instruction in the documentation about how to handle different mip's from `datasets`/`variable` -- I don't agree with you but then again, as long as it works in one way or another then I'm happy(ish) :beer:

Reply:
Let's revert the change for now. We can introduce it in a separate PR later.

Reply:
@valeriupredoi can you revert this change as discussed? We need to finalize this.

Reply:
Change reverted.